hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SupervisorControllerFactory.java

```diff
@@ -0,0 +1,61 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.curator.framework.CuratorFramework;
+import org.glassfish.hk2.api.Factory;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.config.Configs;
+import pl.allegro.tech.hermes.common.di.CuratorType;
+import pl.allegro.tech.hermes.common.metric.HermesMetrics;
+import pl.allegro.tech.hermes.consumers.subscription.cache.SubscriptionsCache;
+import pl.allegro.tech.hermes.consumers.supervisor.ConsumersSupervisor;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.mirror.LegacyMirroringSupervisorController;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.mirror.MirroringSupervisorController;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.ConsumerNodesRegistry;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.selective.SelectiveSupervisorController;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import java.util.Map;
+
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_ALGORITHM;
+
+public class SupervisorControllerFactory implements Factory<SupervisorController> {
+    private final ConfigFactory configs;
```

Comment:
mayby define required configurations as class fields and assign values from ConfigFactory in constructor?